### DISCUSSION TITLE
Единый кабинет селлера на маркетплейсах

### DISCUSSION THREAD

**Discussion by EnzemKa** [totumonline/totum-ru-issues-and-discussions#523] [Сделанные проекты]
Всем здравствуйте. У нас пара интернет-магазинов, мы размещаемся на маркетплейсах с разными способами доставки, товары закупаются у нескольких поставщиков с API или без.
С развитием проекта столкнулись со многими рутинными задачами, решали их через автоматизацию N8N + RetailCRM + Google Sheets. Последние два сервиса, понятное дело, не масштабируемые и не гибкие, что и стало отправной точкой для внедрения Totum.

Текущий стек **N8N + Totum**. Данные, рассчитанные там, передаются в RetailCRM + на сайт(opencart/MySQL) + OZON / YM / MegaMarket / WB / Avito  + оповещения и боты Telegram

Конечно, это не простой проект, который использует только методы Totum, но для нас это освобождает огромное количество времени, которое до этого делали руками. Также, в RetailCRM есть много модулей по каждому МП, можно отдать кучу денег и это заменит нашу схему). Но все это упирается не только в стоимость каждого модуля, но и в их ограничения и возможности

**Какие проблемы решает проект**:
1) Актуальное кол-во товаров на маркетплейсах: при заказе последней единицы товара со склада, в один момент отключаются все Экспрессы, все доставки FBS (с нашего склада), но остаются продажи напрямую от поставщиков. Аналогично при обновлении данных от поставщиков. Остальная информация передается по крону.

2) Автоматический расчет цен относительно закупочных: поставщик передает цену за товар - Тотум рассчитывает цену продажи на сайт + акционную цену на сайт + цену товара для каждой схемы продажи на маркетплейсе с учетом их всех полных комиссий (UNIT-экономика) + наша наценка или скидка.
!! Тотум рассчитывает себестоимость и прибыль с продажи товара на разных схемах продажи (FBS или DBS или Express) и отключает невыгодные схемы продаж

3) Ежедневно N8N мониторит комиссии на всех маркетплейсх, при их обновлении (а они, как оказалось, каждый день изменяют стоимости логистики или возвратов) заново рассчитываются цены товаров на маркетплейсе, который обновил комиссии.

4) Мониторинг РРЦ с учетом акций - если поставщик контролирует, парсит и блокирует отгрузки при снижении цен, то необходимо проверять размещение и вовремя поднимать их, чтобы не улететь в бан

5) Отдельные отчеты по броням товаров в заказах, остатки, данные заказов. Отчеты с проблемными товарами (если акция невыгодная, поставщик передал некорректные данные, и т.д), то автоматически отключаются продажи. Отчет по отслеживаемым товарам (можно отметить товары, у которых нужно мониторить количество).

И много-много других моментов, которые помогает решить Тотум. Причем, чем больше и больше разрабатываем в Тотум, тем больше задач придумываем и реализовываем. Например, ежедневные путевые листы для наших курьеров для предоставления в ГИБДД или налоговую, что вообще не входило в планы, но легко реализовалось :) Я расскажу только про основу нашего проекта.

ГЛАВНАЯ:

Дэшборд с быстрыми ссылками в отчеты или таблицы
Поиск товаров по слову, например, нужно вывести все кондиционеры в таблице - сделали свой, так как в селекте или фильтре невозможно долго тыкать каждый кондиционер, Meili Search тоже не подошел. Решили просто: при добавлении товара разбили название на тэги в поле Данные, и добавили в фильтр.

![1](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/42211689/27212b47-50a1-4a94-8c66-15408ca59687)

ПОЛНЫй КАТАЛОГ ТОВАРОВ:

Это не основная линия, но с этой таблицы начинается работа с товарами. Вся сложность в том, что у товаров есть разные варианты - комплектация или цвет. Так как в МП размещаются именно варианты товаров, то основная линия это следующая таблица - Товарные предложения.

В каталог загружаются по кнопке товары с сайта. Далее автоматически создается для него товарное предложение. И если у товара несколько вариантов, то мы можем создать дополнительное.

![2](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/42211689/02934285-4fe7-47a5-94bb-20e86bc6e8c1)

ТОВАРНЫЕ ПРЕДЛОЖЕНИЯ

Основная линия. Каждое предложение - цикл, в цикле таблица с вариантами размещения на маркетплейсах.

Есть три вида отображения таблицы: Основной, Цены и Остатки
- В основном виде можно менять доступность на сайте и в маркетплейсах, изменять Поставщика, а также добавлять остатки на склад и менять цену товара.

Некоторые поля товара блокируются для изменения, если у него выбран Поставщик с настроеной интеграцией. Для изменения данных необходимо отключить Автообновление.
* При включении автообновления, автоматически проставится актуальная цена = РРЦ и количество товара у поставщика.

- При включении вида Остатки скрываются большинство столбцов для комфортной работы с количеством товара.
- При включении вида Цены отображаются закупочные, акционные и основные цены товарных предложений.

![3](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/42211689/ed689113-8cbe-4a39-a077-23d8eba97d56)

ЦИКЛ Товарного предложения

Основные возможности: автоматический расчет и информация по ценам, настройка схем работы с маркетплейсами, форсированное обновление данных на маркетплейс/сайт (основная синхронизация по крону).

![4 1](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/42211689/0e36a897-847e-4955-afa7-126bac70359b)

Чтобы Тотум знал, какие товарные предложения относятся к товарам из маркетплейсов, их нужно связать. Для этого по API загружаются каталоги каждого МП и они связываются в секции Маркетплейсы через селекты.

![4 мп](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/42211689/01554258-4f5f-4106-84aa-dca7b8854d1d)

Для каждого размещения товара можно настроить откуда брать остатки - например Экспресс с нашего склада, DBS со склада поставщика и др... Также можно фиксировать значение для размещения. На каждую цену есть расчет прибыли.

![4 2](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/42211689/4dc18746-f091-4147-adb6-7411601df3fd)

![4 3](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/42211689/1e81fddf-db03-45bd-b82a-f6327e5e92b8)

СЛАДЫ

Также реализована удобная работа со схемами продаж на маркетплейсах - можно настраивать ручное или API обновление данных. И останавливать продажи из одного окна на всех мп.

![5](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/42211689/3b046c04-876a-41c3-91db-549caae9e2a9)

В целом Тотум нам очень помог: работает быстро, при определенном желании и способностях, можно прикрутить абсолютно все, на что хватает потребностей и фантазии. Спасибо.

**Comment by ttm-alexeypolunin** [comment]
Спасибо, что написали! Выглядит очень аккуратно.

Мне очень понравилась идея использовать иконки `toggle-on` и `toggle-off` для обозначения статуса — забрал в копилку :)

**Reply by ttm-alexeypolunin** [reply]
Если возможно, поделитесь: сколько у вас времени заняло разработать решение от момента, когда приступили до первого запуска в эксплуатацию? И какая была квалификация и размер команды, которая занималась проектом? Спс

**Reply by EnzemKa** [reply]
Да, конечно, команда из одного человека - меня, квалификация "low-code" программист) Наткнулась и начала изучать Тотум прошлым летом, в августе начала тестировать в связке с n8n. Разработка заняла примерно 3 месяца = формулы и расчеты, основная интеграция с системами и поставщиками. Месяц тестирование и отлавливание ошибок. С января постепенно включали синхронизацию с МП, и сразу же перешли на работу в Тотум.

**Comment by LostAlly** [comment]
Большое спасибо, один из лучших опубликованных проектов.
А почему вы остатки вынесли в отдельную временную, а не отображаете их в товарной таблице.

**Reply by EnzemKa** [reply]
Спасибо за отзыв! Перенесли в отдельный отчет для вызова Анонимной таблицы из RetailCRM без авторизации пользователя, чтобы быстро добавлять приход товара на склад. Мы пока не готовы отказаться от той CRM системы, работать с заказами с ней удобнее
